fsthttp/backend.go

```diff
@@ -279,6 +280,19 @@ func (b *BackendOptions) SNIHostname(host string) *BackendOptions {
 	return b
 }
 
+// ClientCertificate sets the client certificate to be provided to the server as part of the SSL handshake.
+func (b *BackendOptions) ClientCertificate(certificate string, key secretstore.Secret) *BackendOptions {
+	b.abiOpts.UseSSL(true)
```

Comment:
Should other SSL-related things (`CertHostname`, `CACert`, `Ciphers`, `SNIHostname`, `SSLMaxVersion`, `SSLMinVersion`) also imply `UseSSL(true)`?  Feels like either all of them should or none of them should, and we should make it clear in the docs if so.

Reply:
This behaviour was copied from the Rust SDK. 
```
    /// Provide the given client certificate to the server as part of the SSL handshake.
    //
    /// Setting this will enable SSL for the connection as a side effect. Both
    /// the certificate and the key to use should be in standard PEM format;
    /// providing the information in another format will lead to an error. We
    /// suggest that (at least the) key should be held in something like the
    /// Fastly secret store for security, with the handle passed to this function
    /// without unpacking it via [`Secret::plaintext`]; the certificate can
    /// be held in a less secure medium.
    ///
    /// (If it is absolutely necessary to get the key from another source, we
    /// suggest the use of [`Secret::from_bytes`]).
    pub fn provide_client_certificate(
        mut self,
        pem_certificate: impl ToString,
        pem_key: Secret,
    ) -> Self {
        self.use_ssl = true;
        self.client_certificate_info = Some((pem_certificate.to_string(), pem_key));
        self
    }
```

The other SSL-related calls also set `use_ssl = true` in the Rust SDK.  I'll update our implementation as well.